src/workers/postUpdated.ts

```diff
@@ -182,22 +197,42 @@ const updatePost = async ({
 
   if (
     !databasePost ||
-    databasePost.metadataChangedAt.toISOString() >= updatedDate.toISOString()
+    databasePost.metadataChangedAt.toISOString() >=
+      data.metadataChangedAt.toISOString()
   ) {
     return null;
   }
 
   const title = data?.title || databasePost.title;
-  const updateBecameVisible = !databasePost.visible && !!title?.length;
+  const updateBecameVisible =
+    content_type === PostType.Freeform
+      ? true
+      : !databasePost.visible && !!title?.length;
 
   data.id = databasePost.id;
-  data.metadataChangedAt = updatedDate;
   data.title = title;
   data.visible = updateBecameVisible;
   data.visibleAt = updateBecameVisible
-    ? databasePost.visibleAt ?? updatedDate
+    ? databasePost.visibleAt ?? data.metadataChangedAt
     : null;
 
+  if (content_type in allowedFieldsMapping) {
+    const allowedFields = [
+      'id',
+      'title',
+      'visible',
+      'visibleAt',
+      'flags',
```

Comment:
Unsure if `title` should be allowed or not

Reply:
Title should not be allowed to be updated in this flow.

Reply:
Also technically the ID should never change here.

Reply:
Removed the title.

Reply:
Yeah, the ID should never change, but to not remove it from the data object, where it is defined a bit further up, I added it to the "always allowed" fields. Same with the rest.
https://github.com/dailydotdev/daily-api/blob/c00f3713688ca1bca848d15bcdbaa2bbc74bfef7/src/workers/postUpdated.ts#L212